android-components/components/feature/media/src/main/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeature.kt

```diff
@@ -35,51 +35,53 @@ class MediaSessionFullscreenFeature(
             flow.map {
                 it.tabs + it.customTabs
             }.map { tab ->
-                tab.filter { it.mediaSessionState != null && it.mediaSessionState!!.fullscreen }
-            }.distinctUntilChanged().collect { states ->
-                processFullscreen(states)
-                processDeviceSleepMode(states)
+                tab.firstOrNull { it.mediaSessionState != null && it.mediaSessionState!!.fullscreen }
+            }.distinctUntilChanged { old, new ->
+                old?.mediaSessionState?.fullscreen == new?.mediaSessionState?.fullscreen &&
+                    old?.mediaSessionState?.playbackState == new?.mediaSessionState?.playbackState &&
+                    old?.content?.pictureInPictureEnabled == new?.content?.pictureInPictureEnabled &&
+                    old?.mediaSessionState?.elementMetadata == new?.mediaSessionState?.elementMetadata
+            }.collect { state -> /* there should only be one fullscreen session */
+                if (state == null || state.mediaSessionState?.fullscreen != true) {
```

Comment:
I don't think the second half of this OR can be hit, since the tab will be mapped to `null` if `fullscreen` is false in `tab.firstOrNull { it.mediaSessionState != null && it.mediaSessionState!!.fullscreen }`

Reply:
That seems totally true! Thanks for noticing that